src/utils.rs

```diff
@@ -0,0 +1,232 @@
+//! Utility functions for the Sprocket command line tool.
+
+use std::fs;
+use std::path::PathBuf;
+
+use anyhow::Context;
+use anyhow::Result;
+use anyhow::bail;
+use tracing::debug;
+
+/// Parse result containing either JSON or YAML data
+#[derive(Debug)]
+pub enum ParsedInput {
+    /// JSON data
+    Json(serde_json::Value),
+    /// YAML data converted to JSON
+    Yaml(serde_json::Value),
+}
+
+/// Parses an input file as either JSON or YAML.
+///
+/// Tries to parse as JSON first, then YAML if JSON parsing fails.
+/// Returns the parsed data as a JSON value.
+pub fn parse_input_file(path: &PathBuf) -> Result<ParsedInput> {
+    let content = fs::read_to_string(path)
+        .with_context(|| format!("Failed to read file: {}", path.display()))?;
+
+    // Try parsing as JSON first
+    if let Ok(json_value) = serde_json::from_str::<serde_json::Value>(&content) {
+        debug!("File parsed as JSON: {}", path.display());
+        return Ok(ParsedInput::Json(json_value));
+    }
+
+    // Then try parsing as YAML
+    if let Ok(yaml_value) = serde_yaml::from_str::<serde_yaml::Value>(&content) {
+        debug!("File parsed as YAML: {}", path.display());
+        // Convert YAML to JSON
+        let json_value = serde_json::to_value(&yaml_value)
+            .context("Failed to convert YAML value to JSON value")?;
+        return Ok(ParsedInput::Yaml(json_value));
+    }
+
+    // If neither worked, return an error
+    bail!(
+        "File is neither valid JSON nor valid YAML: {}",
+        path.display()
+    )
+}
+
+/// Gets a JSON string representation of the input file.
+///
+/// Takes ownership of the input path and parses the file only once.
```

Comment:
```suggestion
```
This information is a matter of the internal implementation, that a user of this func doesn't need to concern themselves with.